projects/solar/src/sense_mcp3427.c

```diff
@@ -0,0 +1,84 @@
+#include "sense_mcp3427.h"
+#include "data_store.h"
+#include "event_queue.h"
+#include "log.h"
+#include "mcp3427_adc.h"
+#include "sense.h"
+#include "solar_events.h"
+
+// We register one sense callback per MCP3427 to take advantage of the sense loop, and so that the
+// entire MCP3427 sense operation doesn't stop if there is one fault in the sense cycle.
+
+// we only use one channel on the MCP3427
+#define SENSE_MCP3427_CHANNEL MCP3427_CHANNEL_1
+
+static uint8_t s_num_mcp3427s;
+static Mcp3427Storage s_mcp3427_storages[MAX_SOLAR_MCP3427];
+static DataPoint s_mcp3427_data_points[MAX_SOLAR_MCP3427];
+static uint8_t s_indices[MAX_SOLAR_MCP3427];  // for contexts
+
+static int16_t s_values[MAX_SOLAR_MCP3427];
+static bool s_has_value[MAX_SOLAR_MCP3427];
+static uint8_t s_consecutive_faults[MAX_SOLAR_MCP3427];
```

Comment:
This was probably here the last time I looked at it, but didn't realize. This is normally handled by having a big struct. And then pass the pointer to the correct struct instead of passing around the index.